Services/Common/OJS.Services.Common.Data/Implementations/SubmissionsForProcessingCommonDataService.cs

```diff
@@ -80,15 +78,23 @@ public async Task AddOrUpdateBySubmissionIds(ICollection<int> submissionIds)
             .Select(sId => new SubmissionForProcessing
             {
                 SubmissionId = sId,
+                Processed = false,
+                Processing = false,
             });
 
-        using var scope = TransactionsHelper.CreateTransactionScope();
+        using var scope = TransactionsHelper.CreateTransactionScope(
+            isolationLevel: IsolationLevel.RepeatableRead,
+            asyncFlowOption: TransactionScopeAsyncFlowOption.Enabled);
```

Comment:
Why you need these?

Reply:
It is due to the following problem: "A TransactionScope must be disposed on the same thread that it was created.".   asyncFlowOption: TransactionScopeAsyncFlowOption.Enabled) fixes this.